src/libYARP_dev/src/devices/ControlBoardWrapper/SubDevice.cpp

```diff
@@ -277,10 +277,10 @@ bool SubDevice::attach(yarp::dev::PolyDriver *d, const std::string &k)
     if(!pos && !pos->getAxes(&subdevAxes))
     {
 
+        attachedF = false;
```

Comment:
This line was never executed before, and it is being moved to a place where it will actually be executed, therefore it is potentially dangerous 😁
@barbalberto @randaz81 can you please review this change?

Reply:
In most cases, `attachedF` was already false. Anyhow, as there are a lot of error checks in this function, and none of those set `attachedF` to false, I would not set it (or set it in all error conditions, but I would go for the less invasive change).

Reply:
Dropped the `attachedF = false` in https://github.com/robotology/yarp/pull/1387/commits/6ab54ea15aceaec2c3ead617f2da737ed6892d3a